src/Plugins/SimplnxCore/src/SimplnxCore/Filters/ReshapeDataArrayFilter.cpp

```diff
@@ -0,0 +1,204 @@
+                                inputArrayPath.toString(), numTuples)});
+      }
+    }
+
+    resultOutputActions.value().appendAction(std::make_unique<CreateNeighborListAction>(inputNeighborList.getDataType(), numTuples, outputArrayPath));
+    break;
+  }
+  case IArray::ArrayType::StringArray: {
+    auto& inputStringArray = dataStructure.getDataRefAs<StringArray>(inputArrayPath);
+    if(tDims.size() > 1)
+    {
+      if(numTuples == inputStringArray.getNumberOfTuples())
+      {
+        return MakePreflightErrorResult(to_underlying(ReshapeDataArray::ErrorCodes::TupleShapesEqual),
+                                        fmt::format("The input array '{}' is a string array, which does not support multiple tuple dimensions. "
+                                                    "The selected tuple shape [{}] cannot be converted to [{}] because it matches the string array's tuple shape ([{}]). "
+                                                    "Please choose a tuple shape that results in a different number of tuples than the string array.",
+                                                    inputArrayPath.toString(), fmt::join(tDims, ","), numTuples, numTuples));
+      }
+      else
+      {
+        resultOutputActions.warnings().push_back(
+            Warning{to_underlying(ReshapeDataArray::WarningCodes::StringArrayMultipleTupleDims),
+                    fmt::format("The input array '{}' is a string array, and string arrays do not support multiple tuple dimensions.  The string array will be reshaped to [{}] instead.",
+                                inputArrayPath.toString(), numTuples)});
+      }
+    }
+
+    resultOutputActions.value().appendAction(std::make_unique<CreateStringArrayAction>(tDims, outputArrayPath));
+    break;
+  }
+  case IArray::ArrayType::Any: {
+    return MakePreflightErrorResult(to_underlying(ReshapeDataArray::ErrorCodes::InputArrayEqualsAny),
+                                    fmt::format("Input array '{}' has array type 'Any'.  Something has gone horribly wrong, please contact the developers.", inputArray.getName()));
+  }
+  default: {
+    return MakePreflightErrorResult(to_underlying(ReshapeDataArray::ErrorCodes::InputArrayUnsupported),
+                                    fmt::format("Input array '{}' has array type '{}'.  This array type is not currently supported by this filter, please contact the developers.",
+                                                inputArray.getName(), inputArray.getTypeName()));
+  }
+  }
+
+  resultOutputActions.value().appendDeferredAction(std::make_unique<DeleteDataAction>(inputArrayPath));
+  resultOutputActions.value().appendDeferredAction(std::make_unique<RenameDataAction>(outputArrayPath, inputArrayPath.getTargetName()));
+
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
```

Comment:
```suggestion
  return {std::move(resultOutputActions)};
```
Just use overload that constructs via singular value to avoid carrying extra weight during preflight since it runs at least 3 times before execution